builtin/plugin/backend.go

```diff
@@ -71,8 +77,8 @@ func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 	return &b, nil
 }
 
-// backend is a thin wrapper around plugin.BackendPluginClient
-type backend struct {
+// PluginBackend is a thin wrapper around plugin.BackendPluginClient
+type PluginBackend struct {
```

Comment:
Does this have to be exported? The `func Backend` factory func above returns an interface so having this as `backend` seems to be fine (and follows the pattern on how other backends are laid out, e.g. `credentials/aws` having its own internal `type backend struct`).

Reply:
This is exported so it can be used in the `plugin_test` package residing in the same directory, [here](https://github.com/hashicorp/vault/pull/5536/files#diff-cd8f86ab1420ec95c1ed441097f5a543R22). 

The reason that was changed to its own package was to resolve an extremely persistent import cycle. This is the import cycle that results if I revert it:
```